File: Crack.py
import sys

def encrypt(fileName,keyword):
	fin = open(fileName,'r')
	s = fin.read()
	arr = []
	for i in xrange(0,len(s)):
		arr.append(ord(s[i])^ord(keyword[i%3]))
	fin.close()
	fout = open(fileName,'w');
	for j in arr:
		fout.write(str(j)+",");
	return
	
def decrypt(fileName,keyword,guess):
	if guess:
		fin = open(fileName,'r')
		s = fin.read()
		arr = s.split(',')
		print "<-----------------------wild guess------------------------>"
		for i in xrange(0,len(arr)-1):
			sys.stdout.write(chr(int(arr[i])^ord(keyword[i%3]))),
		print "\n<--------------------------------------------------------->"
		fin.close()
	else:
		fin = open(fileName,'r')
		s = fin.read()
		arr = s.split(',')
		out = ""
		for i in xrange(0,len(arr)-1):
			out += chr(int(arr[i])^ord(keyword[i%3]))
		fin.close()
		fout = open(fileName,'w')
		fout.write(out);
		fout.close()
	return
	
if __name__=='__main__':
	if sys.argv[1] == "encrypt":
		if len(sys.argv) < 3 :
			encrypt(sys.argv[2],"yup")
		else:
			encrypt(sys.argv[2],sys.argv[3])
	elif sys.argv[1] == "decrypt":
		if(len(sys.argv)<5):
			decrypt(sys.argv[2],sys.argv[3],False)
		elif(sys.argv[3] == "-guess"):
			decrypt(sys.argv[2],sys.argv[4],True)

File: Crack.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
using namespace std;
void encrypt(char* fileName,string key);
void decrypt(char* fileName,string key,bool guess);

int main(int argc,char* argv[]){
	if(argc < 2){
		cout << "Wrong input format for:\n\tCrack <operation> <file_name> <flag> <key>" << endl;
	}
	if(string(argv[1]) == "encrypt"){
		if(argc < 4){
			encrypt(argv[2],string("yup"));
		}else{
			encrypt(argv[2],argv[3]);
		}
	}else if(string(argv[1]) == "decrypt"){
		if(argc < 4){
			cout << "Secret key is required to decrypt!" << endl;
		}else{
			if(string(argv[3]) == "-guess"){
				decrypt(argv[2],string(argv[4]),true);
			}else{
				decrypt(argv[2],string(argv[3]),false);
			}
		}
	}else{
		cout << "Unknown operation. Operation should be either encrypt or decrypt! " << endl;
	}
}
void encrypt(char* fileName,string key){
	ifstream fin(fileName);
	string line;
	int a,i,j;
	char c[3];
	c[0] = key.at(0);	c[1]=key.at(1);	c[2]=key.at(2);
	vector<int> arr;
	if(fin.is_open()){
		j=0;
		while(getline(fin,line)){
			cout << line << endl;
			for(i=0;i<line.length();i++){
				arr.push_back(line.at(i)^c[j%3]);
				j++;
			}
		}
	}
	fin.close();
	ofstream fout(fileName);
	if(fout.is_open()){
		for(vector<int>::iterator it = arr.begin();it != arr.end();it++){
			fout << *it << ",";
		}
	}
	fout.close();
}
void decrypt(char* fileName,string key,bool guess){
	ifstream fin(fileName);
	string line;
	int a,i;
	char c[3];
	c[0] = key.at(0);	c[1]=key.at(1);	c[2]=key.at(2);
	vector<char> arr;
	if(fin.is_open()){
		i = 0;
		if(guess){
			cout << "<-------------------------wild guess------------------------>" << endl;
			while(fin >> a){
				cout << (char)(a^c[i%3]);
				if(fin.peek() == ','){
					fin.ignore();
				}
				i++;
			}
			cout << endl << "<----------------------------------------------------------->" << endl;
		}else{
			while(fin >> a){
				arr.push_back((char)(a^c[i%3]));
				if(fin.peek() == ','){
					fin.ignore();
				}
				i++;
			}
		}
	}
	fin.close();
	if(!guess){
		ofstream fout(fileName);
		if(fout.is_open()){
			for(vector<char>::iterator it = arr.begin();it != arr.end();it++){
				fout << *it;
			}
		}
	}
} 